unityfiles/Assets/Scripts/Game/Weapon/GunController.cs

```diff
@@ -193,23 +207,29 @@ public class GunController : MonoBehaviour {
 
     // -- unity -- //
 
-    void Start() {
+    void Awake() {
         //TODO: Discuss system for timer naming conventions
-        this.SetBulletWaitTime(this.fireRate);
+        this.SetBulletWaitTime(this.FireRate);
         this.timerUID = this.fireRateTimer.RollingUID;
         this.fireRateTimer.Set(this.timerUID, bulletWaitTime);
-        this.GenerateBulletBlueprint();
     }
 
     void Update() {
-        if (Input.GetMouseButtonDown(0)) {
-            this.MaybeFire();
-        }
+        // todo: This meens the fire rate is capped by the framerate this may be a non issue
         if (isFiring) {
             this.MaybeFire();
         }
     }
 
+    /// <summary>
+    /// This function is called when the MonoBehaviour will be destroyed.
+    /// </summary>
+    void OnDestroy() {
+        foreach (GameObject GO in activeBullets) { Destroy(GO); }
+        foreach (GameObject GO in idleBullets) { Destroy(GO); }
+        Destroy(blueprintBullet);
+    }

```

Comment:
Husk at dette tar livet av alle aktive kuler når man skifter våpen; derav finne en måte å la vær å destroye controlleren

Reply:
saken er at det er egentlig ikke noen grunn til og bytte ut gun controlleren slik jeg jør. planen er at på en varm sommerdag i fremmtiden en gang skal jeg flippe den omm slik at man bare opdaterer hva som skal skytes fra nå av istedet for og fjerne og igjenn lage en GC.